### DISCUSSION TITLE
Instrumentation of Open Telemetry with .net based Microservices for Prometheus, Loki and Tempo and visualize in Grafana.

### DISCUSSION THREAD

**Discussion by raghava28041978** [open-telemetry/community#1853] [Community Discussions]
Is there documentation on how to integrate Open Telemetry with .net based Microservices to gather Metrics, logs, traces etc using  prometheus, Loki and Tempo and visualize in Grafana. I am new to Grafana and rest of the components. The Microservices are installed on the Azure Kubernetes Cluster and we would like to get dashboards that provide same visualization and flexibility as Log analytics in Azure by using Grafana. We would like to do away with Log Analytics due to higher costs. Any help around this is much appreciated.

**Comment by svrnm** [comment]
Hey @raghava28041978,

the OpenTelemetry documentation can help you with the integration of traces, metrics logs into your .NET based microservices and if needed the use of the opentelemetry operator & collector:

https://opentelemetry.io/docs/instrumentation/net/
https://opentelemetry.io/docs/collector/
https://opentelemetry.io/docs/kubernetes/operator/

Those docs should get you to a point where you can export logs, metrics traces either via OTLP, or via dedicated exporters, e.g. prometheus, & loki.

From there you should reference the documentations provided by grafana to understand how you can make sense out of that data collected.

**Reply by raghava28041978** [reply]
Thanks a lot Severin for the quick reply!!  I will explore the documentation and come back here in case of further assistance.

**Comment by pyohannes** [comment]
Hi @raghava28041978,

Here are some Grafana-specific links in addition to the OTel ones given:
* If you want to use Grafana Cloud, have a look at its [Kubernetes](https://grafana.com/solutions/kubernetes/) and [Application Observability](https://grafana.com/docs/grafana-cloud/monitor-applications/application-observability/) plugins. For the latter, there's a [.NET quick start guide](https://grafana.com/docs/grafana-cloud/monitor-applications/application-observability/setup/quickstart/dotnet/).
* If you want to host the Grafana stack yourself, start playing with the [docker-otel-lgtm](https://github.com/grafana/docker-otel-lgtm) project, that packs the whole Grafana stack into a single docker image and configures it for ingesting OTLP data. This will not scale for production scenarios, but can serve as a starting point for experiments.
* For Grafana-specific questions, feel free to reach out on the public Grafana [#opentelemetry](https://grafana.slack.com/archives/C038S86TZGF) slack channel.
